plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/StmtConversionContext.kt

```diff
@@ -134,18 +132,31 @@ fun StmtConversionContext.embedPropertyAccess(accessExpression: FirPropertyAcces
             error("Property access symbol $calleeSymbol has unsupported type.")
     }
 
+
+fun StmtConversionContext.argumentDeclaration(arg: ExpEmbedding, callType: TypeEmbedding): Pair<Declare?, ExpEmbedding> =
+    when (arg.ignoringMetaNodes()) {
+        is LambdaExp -> null to arg
+        else -> {
+            val argWithInvariants = arg.withNewTypeInvariants(callType) {
+                proven = true
+                access = true
+            }
+            if (argWithInvariants is VariableEmbedding) null to argWithInvariants
+            else declareAnonVar(callType, argWithInvariants).let {
+                it to it.variable
+            }
```

Comment:
My impression was that we already make copies of values passed to inlined functions; otherwise the following code is not translated correctly:

```kotlin
class X(var a: Int)
fun f(x: X) : Int = x.a.run { x.a += 1; it }
```
This isn't an idiomatic example, of course, but do we miscompile it?

Reply:
Hm, I added a couple of tests now that check we don't evaluate receiver twice. I'm not able to find any other copies when passing arguments. Anyway this part is just slightly changed logic of what was in this file before.

Reply:
I think I'll try to experiment without this piece of code and report what happens.